deepinterpolation/generator_collection.py

```diff
@@ -897,112 +900,144 @@ class OphysGenerator(SequentialGenerator):
 
         self._update_end_frame(self.total_frame_per_movie)
         self._calculate_list_samples(self.total_frame_per_movie)
 
-        average_nb_samples = np.min([int(raw_data.shape[0]), 100])
-
-        # For backward compatibility
-        if "cache_data" in self.json_data.keys():
-            self.cache_data = self.json_data["cache_data"]
-        else:
-            self.cache_data = False
-
-        if self.cache_data:
-            print('Caching hdf5 file... \n')
-            if self.end_frame > 0:
-                self.raw_data = raw_data[0:np.min([self.total_frame_per_movie,
-                    self.end_frame+self.post_frame+self.pre_post_omission+1]), :, :]
-            else:
-                self.raw_data = raw_data[:, :, :]
-            local_data = self.raw_data[0:average_nb_samples, :, :].flatten()
-        else:
-            local_data = raw_data[0:average_nb_samples, :, :].flatten()
-            self.cache_data = False
-
+        average_nb_samples = np.min([int(self.movie_data.shape[0]), 1000])
+        local_data = self.movie_data[:average_nb_samples].flatten()
         local_data = local_data.astype("float32")
 
         self.local_mean = np.mean(local_data)
         self.local_std = np.std(local_data)
-        
-        if self.cache_data:
-            self.raw_data = ( self.raw_data.astype("float") \
-                - self.local_mean ) / self.local_std
-
-        movie_obj_point.close()
-
-    def __getitem__(self, index):
-        shuffle_indexes = self.generate_batch_indexes(index)
-
-        input_full = np.zeros(
-            [self.batch_size, 512, 512, self.pre_frame + self.post_frame],
-            dtype="float32",
-        )
 
-        output_full = np.zeros([self.batch_size, 512, 512, 1], dtype="float32")
-
-        for batch_index, frame_index in enumerate(shuffle_indexes):
-            X, Y = self.__data_generation__(frame_index)
-            X_shape = X.shape
-            Y_shape = Y.shape
-
-            input_full[batch_index, : X_shape[0], : X_shape[1], :] = X
-            output_full[batch_index, : Y_shape[0], : Y_shape[1], 0] = Y
-
-        return input_full, output_full
-
-    def __data_generation__(self, index_frame):
-        "Generates data containing batch_size samples"
-
-        if self.cache_data:
-            movie_obj = self.raw_data
+        if self.gpu_available:
+            self.data_tensor = False
+            if self.gpu_cache_full:
+                print("Caching full movie onto GPU")
+                self.data_tensor = tf.convert_to_tensor(
+                    self.movie_data, dtype="float")
+                self.data_tensor = (self.data_tensor - self.local_mean) / self.local_std
+            else:
+                # Get indices for partial movie cache
+                start = self.pre_frame + self.pre_post_omission
+                end = start + self.batch_size
+                self.batch_indices = list(range(start,end))
```

Comment:
I wanted to set this instance variable because the input_indices and batch_indices are constant across all batches when slicing from the chunked movie. If it's set in __get_item__, it would regenerate the batch_indices every time it's called, which would be unnecessary compute.